src/main/java/sg/ncl/MainController.java

```diff
@@ -1331,14 +1357,15 @@ public String contributeData(Model model) {
     }
     
     @RequestMapping(value="/data/contribute", method=RequestMethod.POST)
-    public String validateContributeData(@ModelAttribute("dataset") Dataset dataset, HttpSession session, @RequestParam("file") MultipartFile file, RedirectAttributes redirectAttributes) {
+    public String validateContributeData(@ModelAttribute("dataset") Dataset dataset, HttpSession session, @RequestParam("file") MultipartFile file, RedirectAttributes redirectAttributes) throws IOException {
     	// TODO
     	// validation
     	// get file from user upload to server
+        BufferedOutputStream stream = null;
 		if (!file.isEmpty()) {
 			try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this "try" to a try-with-resources. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2093)